[Bug] [Master] Create WorkflowExecuteRunnable failed

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

[WI-0][TI-0] - [ERROR] 2024-03-06 08:52:31.137 +0800 o.a.d.s.m.r.MasterSchedulerBootstrap:[156] - Master handle command 2007 error 
org.apache.dolphinscheduler.server.master.exception.WorkflowCreateException: Create WorkflowExecuteRunnable failed
	at org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteRunnableFactory.createWorkflowExecuteRunnable(WorkflowExecuteRunnableFactory.java:98)
	at org.apache.dolphinscheduler.server.master.runner.MasterSchedulerBootstrap.lambda$run$0(MasterSchedulerBootstrap.java:139)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:291)
	at java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:731)
	at java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:289)
	at java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:401)
	at java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:734)
	at java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:160)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:583)
	at org.apache.dolphinscheduler.server.master.runner.MasterSchedulerBootstrap.run(MasterSchedulerBootstrap.java:136)
Caused by: java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.setGlobalParamIfCommanded(ProcessServiceImpl.java:604)
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.generateNewProcessInstance(ProcessServiceImpl.java:561)
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.constructProcessInstance(ProcessServiceImpl.java:707)
	at org.apache.dolphinscheduler.service.process.ProcessServiceImpl.handleCommand(ProcessServiceImpl.java:291)


### What you expected to happen

在执行“set start param into global params”逻辑时，需要判空处理

![image](https://github.com/apache/dolphinscheduler/assets/13687890/0ad8ec4c-0cb2-4602-80d7-9bcd1b305618)


### How to reproduce

工作流定义一个全局参数，类型为OUT，值为空，然后正常执行任务

![image](https://github.com/apache/dolphinscheduler/assets/13687890/7de8a69a-9dd5-4fcd-be1d-93485b61c24a)


### Anything else

_No response_

### Version

3.2.x

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java
+++ b/dolphinscheduler-service/src/main/java/org/apache/dolphinscheduler/service/process/ProcessServiceImpl.java
@@ -601,9 +601,12 @@ public class ProcessServiceImpl implements ProcessService {
         if (MapUtils.isNotEmpty(startParamMap) && globalMap != null) {
             // start param to overwrite global param
             for (Map.Entry<String, String> param : globalMap.entrySet()) {
-                String val = startParamMap.get(param.getKey()).getValue();
-                if (val != null) {
-                    param.setValue(val);
+                String globalKey = param.getKey();
+                if (startParamMap.containsKey(globalKey)) {
+                    String val = startParamMap.get(globalKey).getValue();
+                    if (val != null) {
+                        param.setValue(val);
+                    }
                 }
             }
             // start param to create new global param if global not exist

--- a/dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/process/ProcessServiceTest.java
+++ b/dolphinscheduler-service/src/test/java/org/apache/dolphinscheduler/service/process/ProcessServiceTest.java
@@ -71,11 +71,13 @@ import org.apache.dolphinscheduler.dao.repository.TaskDefinitionDao;
 import org.apache.dolphinscheduler.dao.repository.TaskDefinitionLogDao;
 import org.apache.dolphinscheduler.dao.repository.TaskInstanceDao;
 import org.apache.dolphinscheduler.plugin.task.api.TaskPluginManager;
+import org.apache.dolphinscheduler.plugin.task.api.enums.Direct;
 import org.apache.dolphinscheduler.plugin.task.api.enums.dp.DataType;
 import org.apache.dolphinscheduler.plugin.task.api.enums.dp.DqTaskState;
 import org.apache.dolphinscheduler.plugin.task.api.enums.dp.ExecuteSqlType;
 import org.apache.dolphinscheduler.plugin.task.api.enums.dp.InputType;
 import org.apache.dolphinscheduler.plugin.task.api.enums.dp.OptionSourceType;
+import org.apache.dolphinscheduler.plugin.task.api.model.Property;
 import org.apache.dolphinscheduler.plugin.task.api.model.ResourceInfo;
 import org.apache.dolphinscheduler.service.cron.CronUtilsTest;
 import org.apache.dolphinscheduler.service.exceptions.CronParseException;
@@ -89,6 +91,7 @@ import java.util.Collections;
 import java.util.Date;
 import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.Test;
@@ -610,6 +613,32 @@ public class ProcessServiceTest {
         list.add(dqExecuteResult);
 
         return list;
+    }
+
+    @Test
+    public void testSetGlobalParamIfCommanded() {
+        ProcessDefinition processDefinition = new ProcessDefinition();
+        String globalParams =
+                "[{\"prop\":\"global_param\",\"value\":\"4\",\"direct\":\"IN\",\"type\":\"VARCHAR\"},{\"prop\":\"O_ERRCODE\",\"value\":\"\",\"direct\":\"OUT\",\"type\":\"VARCHAR\"}]";
+        processDefinition.setGlobalParams(globalParams);
+        Map<String, String> globalParamMap = processDefinition.getGlobalParamMap();
+        Assertions.assertTrue(globalParamMap.size() == 2);
+        Assertions.assertTrue(processDefinition.getGlobalParamList().size() == 2);
+
+        HashMap<String, String> startParams = new HashMap<>();
+        String expectValue = "6";
+        startParams.put("global_param", expectValue);
+        HashMap<String, String> commandParams = new HashMap<>();
+        commandParams.put(CMD_PARAM_START_PARAMS, JSONUtils.toJsonString(startParams));
+        Map<String, Property> mockStartParams = new HashMap<>();
+
+        mockStartParams.put("global_param", new Property("global_param", Direct.IN,
+                org.apache.dolphinscheduler.plugin.task.api.enums.DataType.VARCHAR, startParams.get("global_param")));
+        when(curingGlobalParamsService.parseWorkflowStartParam(commandParams)).thenReturn(mockStartParams);
+
+        processService.setGlobalParamIfCommanded(processDefinition, commandParams);
+        Assertions.assertTrue(globalParamMap.get("global_param").equals(expectValue));
+        Assertions.assertTrue(globalParamMap.containsKey("O_ERRCODE"));
     }
 
     @Test

